### ISSUE TITLE
LabelList TypeScript Data Attribute Error

### ISSUE THREAD

**Issue by aexei** [recharts/recharts#2396]
- [X] I have searched the [issues](https://github.com/recharts/recharts/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/recharts-issue-template-r6f5j)

### Steps to reproduce
Create a BarChart and a LabelList within a Bar

### What is expected?
Should compile without error

### What is actually happening?
Throwing typescript error:

Type '{ dataKey: string; position: string; }' is not assignable to type 'IntrinsicAttributes & Pick<Props<Data>, "data" | "id" | "dataKey" | "content" | "clockWise" | "textBreakAll"> & Partial<Pick<Props<Data>, "valueAccessor">> & Partial<...>'.

 Property 'position' does not exist on type 'IntrinsicAttributes & Pick<Props<Data>, "data" | "id" | "dataKey" | "content" | "clockWise" | "textBreakAll"> & Partial<Pick<Props<Data>, "valueAccessor">> & Partial<...>'.ts

(2322)

| Environment | Info |
|---|---|
| Recharts | v2.0.3 |
| React | 16.12.0 |
| System | macOS BigSur 11.1 |
| Browser | Chrome Version 87.0.4280.141 |

**Comment by raviqqe** [comment]
I also have `Label` component's prop type bug where `angle` prop cannot be passed. 

It looks like that there are some old files included in the `types` directory of the npm bundle.

```
> ls types/src
cartesian  chart  component  container  index.d.ts  numberAxis  polar  shape  util

> ls types
cartesian  component  demo        numberAxis  polyfill.d.ts  src
chart      container  index.d.ts  polar       shape          util
```

**Comment by anteburazer** [comment]
I have the same problem with `LabelList ` and `position` prop, when using Recharts with Typescript.
I'm not able to have the labels beside the bar chart

**Comment by hangler** [comment]
Same problem. In my case I have to put `@ts-ignore` **above** the component tag:

```JSX
{/* @ts-ignore The typings are incorrect */}
<LabelList dataKey={'A'} position={'right'} />
```

**Comment by xile611** [comment]
Fixed in v2.0.7
